spacy/tokens/_dict_proxies.py

```diff
@@ -43,6 +43,11 @@ def copy(self, doc: Optional["Doc"] = None) -> "SpanGroups":
             doc = self._ensure_doc()
         return SpanGroups(doc).from_bytes(self.to_bytes())
 
+    def setdefault(self, key, default=None):
+        if not isinstance(default, SpanGroup):
+            raise ValueError("SpanGroups.setdefault only supports default values with type SpanGroup, received type '{default_type}'.".format(default_type=type(default)))
```

Comment:
Add message to `errors.py`?

Also I'd add in a "but" after the comma to make the message even more crystal clear